src/UI/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -44,18 +45,20 @@ internal class ExternalLoginModel<TUser> : ExternalLoginModel where TUser : clas
     {
         private readonly SignInManager<TUser> _signInManager;
         private readonly UserManager<TUser> _userManager;
-        private readonly IUserFactory<TUser> _userFactory;
+        private readonly IUserStore<TUser> _userStore;
+        private readonly IUserEmailStore<TUser> _emailStore;
         private readonly ILogger<ExternalLoginModel> _logger;
 
         public ExternalLoginModel(
             SignInManager<TUser> signInManager,
             UserManager<TUser> userManager,
-            IUserFactory<TUser> userFactory,
+            IUserStore<TUser> userStore,
```

Comment:
code smell doesn't go away just because its internal :)  

I thought we agreed that there was no supported scenario where they would be using the built in pages with a custom user, that the expectation is that they MUST override these with a custom user.  If that's the case, can't we just constrain these pages to be IdentityUser?

Reply:
So I think we can just check in the ctor and throw if TUser isn't derived from `IdentityUser<TKey>` and you can say they need to override the model in this case.

Then we can just set the properties via reflection